plugins/webhooks/github.js

```diff
@@ -226,6 +226,60 @@ function pullRequestEvent(request, reply) {
 }
 
 /**
+ * Act on a Push event
+ *  - Should start a new main job
+ * @method pushEvent
+ * @param  {Hapi.request}       request Request from user
+ * @param  {Hapi.reply}         reply   Reply to user
+ */
+function pushEvent(request, reply) {
+    const pipelineFactory = request.server.app.pipelineFactory;
+    const jobFactory = request.server.app.jobFactory;
+    const buildFactory = request.server.app.buildFactory;
+    const eventId = request.headers['x-github-delivery'];
+    const payload = request.payload;
+    const repository = hoek.reach(payload, 'repository.ssh_url');
+    const branch = hoek.reach(payload, 'ref').replace(/^refs\/heads\//, '');
+    const sha = hoek.reach(payload, 'after');
+    const username = hoek.reach(payload, 'sender.login');
+    const scmUrl = `${repository}#${branch}`;
+    const apiUri = request.server.info.uri;
+
+    request.log(['webhook-github', eventId], `Push for ${scmUrl}`);
+
+    // Fetch the pipeline associated with this hook
+    return pipelineFactory.get({ scmUrl })
+        .then(pipeline => {
+            if (!pipeline) {
+                throw boom.notFound('Pipeline does not exist');
+            }
+
+            // sync the pipeline to get the latest jobs
+            return pipeline.sync()
```

Comment:
I mean pipeline.sync already creates the main job. Why do we do buildFactory.create main again?